code/modules/surgery/robolimbs.dm

```diff
@@ -132,7 +132,7 @@
 	if(..())
 		var/obj/item/robot_parts/p = tool
 		if(p.part)
-			if(!(target_zone in p.part))
+			if(!(affected.body_part in p.part))
```

Comment:
Oh, it's because you changed it here. You can use affected.name instead and use the proper BODY_ZONE defines.